src/diff.rs

```diff
@@ -278,6 +283,21 @@ impl<'repo> Diff<'repo> {
         Ok(buf)
     }
 
+    /// Create an patchid from a diff.
+    pub fn patchid(&self, opts: Option<&mut DiffPatchidOptions>) -> Result<Oid, Error> {
```

Comment:
Underlying `self.raw` is `*mut`, and `raw::git_diff_patchid` takes `*mut`.
`Diff::format_email` choose `&mut self` in similar situation, while `Diff::stats` choose `&self`.
I choose `&self` since it is semantically more natural (`patchid` doesn't seem to modify the diff).
I also wanted to make `opts` to be `Option<&DiffPatchidOptions>` by similar reason (who expects an option to be mutated?), but many of existing other `opts` are `Option<&mut ...>` and it doesn't seem to be easy since the underlying raw value is not a pointer.

Reply:
I think this is fine for now. It largely depends on whether callbacks are involved which they aren't here yet, and this can always be updated later if necessary.